src/properties/propertyowner.cpp

```diff
@@ -111,6 +111,47 @@ Property* PropertyOwner::property(const std::string& uri) const {
     }
 }
 
+PropertyOwner* PropertyOwner::propertyOwner(const std::string& uri) const {
+    PropertyOwner* directChild = propertySubOwner(uri);
+    if (directChild) {
+        return directChild;
+    }
+
+    // If we do not own the searched PropertyOwner, it must consist of a concatenated
+    // name and we can delegate it to a subowner
+    const size_t ownerSeparator = uri.find(URISeparator);
+    if (ownerSeparator == std::string::npos) {
+        // if we do not own the PropertyOwner and there is no separator, it does not exist
+        return nullptr;
+    }
+    else {
+        const std::string parentName = uri.substr(0, ownerSeparator);
+        const std::string ownerName = uri.substr(ownerSeparator + 1);
+
+        PropertyOwner* owner = propertySubOwner(parentName);
+        if (!owner) {
+            return nullptr;
+        }
+        else {
+            // Recurse into the subOwner
+            return owner->propertyOwner(ownerName);
+        }
+    }
+}
+
+std::string PropertyOwner::uri() const {
```

Comment:
Can use this function in `Property::fullyQualifiedIdentifier` now by getting the `uri()` of the owner and adding the property identifier after